urllib3/connectionpool.py

```diff
@@ -438,6 +452,8 @@ def is_same_host(self, url):
         # TODO: Add optional support for socket.gethostbyname checking.
         scheme, host, port = get_host(url)
 
+        host = self._ipv6_host(host)
```

Comment:
No problem on the first request, the second one has a minimal test now (see test/test_connectionpool.py)